src/gmt_nc.c

```diff
@@ -1029,6 +1032,8 @@ GMT_LOCAL int gmtnc_grd_info (struct GMT_CTRL *GMT, struct GMT_GRID_HEADER *head
 
 		/* Define z variable. Attempt to remove "scale_factor" or "add_offset" when no longer needed */
 		gmtnc_put_units (ncid, z_id, header->z_units);
+		if (GMT->parent->remote_info && GMT->parent->remote_id != GMT_NOTSET && GMT->parent->remote_info[GMT->parent->remote_id].CPT[0] != '-')	/* Subset of remote grid with default CPT, save name as an attribute */
+			gmt_M_err_trap (nc_put_att_text (ncid, z_id, "cpt", strlen (GMT->parent->remote_info[GMT->parent->remote_id].CPT), GMT->parent->remote_info[GMT->parent->remote_id].CPT));
```

Comment:
"cpt" is a very GMT parlance. I would prefer "palette" like those grids from OceanColor

Reply:
I don't mind calling it palette, but since it is just a string with a GMT CPT name it is still very GMT centric.  We are not sticking a 3xn array into the grid, just a string.  Still want to rename it?

Reply:
Ok, sorry. I thought that it was the cpt that was added to the nc file and that would be an interesting addition. If it's just a *cpt name suggestion for GMT scripts* then I'm not even sure we should be doing this.

Reply:
Nothing to do with GMT scripts though.  This is for the module and hence for wrappers as well if interested.  Since I think you pass wrapped GMT grids in Julia then this will work for you off the bat.  PyGMT will need to do some other work since they read in a netcdf grid via Xarrays, so they have access to the attributes that way.

We could of course stick a CPT into the netCDF but it has little value in GMT (we would do it because the outside world wants it) and it is limiting in that 3xn matrices do not tell us continuous or discrete and now we have to add more stuff to explain that thing.

Reply:
I am not exploring the hidden struct because there is no guaranties that it won't change and then each change would mean a Julia breakage. I've pulled out some hairs in the past trying to inventing a system where the structs could have some self-referencing system that could be used to find the struct members of interest regardless of structs changes. It would involve the use of ``offsetof`` C macro. But never came to get any workable prototype.

Reply:
OK.  Here is the issue for users in GMT.jl:

```
G = grdcut("@earth_relief_01d_g", region=[0 10 0 10]);
grdimage(G, proj=:Winkel, colorbar=true, show=true)
```

You are getting the turbo colors instead of geo.  Same in GMT master.  But with this PR it has that information available in the subset grid and can be acted upon.  The modules can do it since they either read the grid into the full structure or they pass the full structure as a virtual file to another module.

Another thing we could do, if it made sense for Julia and Python and Matlab, would be to set the current CPT when we **grdcut** from a remote grid.  Then, the above Julia lines would also work as is, and if you really wanted turbo then you would specify it.

Reply:
I.e., we set the default CPT even though we are not plotting, because **grdcut** is often used to select a smaller chunk before plotting.  No harm done if then used for **grdmath**...

Reply:
GMT.jl tries expand the classic mode to work like *modern* and only misses that goal for the modules that require *modern* (*i.e.* ``subplot``). So, for example the, implementing the current cpt concept is achieved via global variable. It's not completely idiotic to imagine that each grid could have a *suggested cpt* (or *none*) but this requires changing the ``GMTgrid`` type (structure). However, I return to the same point. I can't know about it from GMT because the info is stored in the hidden struct. Don't know how PyGMT will find this information using a external tool. Ofc, we can scan fir it from all of the nc attributes, but this seems overkilling.

Reply:
Sorry I see I posted a few comments to the wrong conversation (below).  But if you are not creating a modern session then there is no place to write the current CPT to.  And I am not sure this will help for PyGMT either @meghanrjones since each GMT module call is wrapped in a Create/Destroy session.  So how is the concept of current CPT maintained in PyGMT?

Reply:
BTW, will this work in **classic**?
```
grdcut("@earth_relief_01d_g", region=[0 10 0 10], G="cut.nc");
grdimage("cut.nc", proj=:Winkel, colorbar=true, show=true)
```

Reply:
For CLI and C interface it works in classic since the module can use the full grid header.
For externals needing the current CPT it only works for modern since there is no session in classic and hence no place to write the CPT...

Reply:
Since you are doing stuff under the hood. presumably you could do something when **grdcut** is called? In your C-wrap do you have access to the full header under the hood?  Even if there is a void *hidden pointer? We could imagine adding an API function that could report something based on that void *.

Reply:
If it will work with CLI then it should probably work in Julia (the disk file version) but possibly not because GMT.jl classic sticks in a default CPT in that grdimage command when not CPT is used.

Reply:
Yes, if you do not pass any default CPT but let GMT control the default CPT then it would work.  Are you supplying a different default CPT than GMT would normally (turbo?).  if not then just don't supply one.

Reply:
I think this is a worthwhile improvement for the CLI, regardless of usability for the wrappers. While PyGMT can easily access the cpt attr, I'm not sure how it would communicate it back through the API during the `GMT_Create_Data`, `GMT_Put_Matrix`, `GMT_Open_VirtualFile` process.

[GMT_Set_Comment](https://docs.generic-mapping-tools.org/dev/devdocs/api.html#gmt-set-comment) is only for the 'top-level' grid header (those listed [here](https://docs.generic-mapping-tools.org/dev/devdocs/api.html#gmt-grids)), correct? Could anything be added to the API for setting the items that are mostly used internally (i.e., full list [here](https://docs.generic-mapping-tools.org/dev/devdocs/api.html#gmt-grid))?

Reply:
I don't think that is the step you will want to intervene.  If you learn there is a cpt attribute then you add "-C<cpt>" to the string you prepare for the module, no?

Reply:
Yes, we can do that. If that is the case, we may want to eventually document that users can add a `cpt` attr to their netcdf grid in order to associate a default colormap with their own grids. In which case Joaquim's comment about using ``palette`` over ``cpt`` becomes relevant again.

Reply:
I suppose **grdedit** is the module where it makes sense for us to add an attribute to the grid file. Something like

`grdedit my.grd -Qelevation`

This brings up the issue of "I want my specially made cpt file to be embedded in the grid", to which I suggest someone read the gmtmex_parser.c on the gymnastics on converting a very rich CPT standard into a very poor Matlab color table scheme.  It would be the same issue - lots of work to accommodate all the special things in a CPT, vs just adding the name of the cpt.